controllers/keystoneapi_controller.go

```diff
@@ -880,6 +880,21 @@ func (r *KeystoneAPIReconciler) reconcileNormal(
 	// Create ConfigMaps and Secrets required as input for the Service and calculate an overall hash of hashes
 	//
 
+	// ensure MariaDBAccount exists.  This account record may be created by
+	// openstack-operator or the cloud operator up front without a specific
+	// MariaDBDatabase configured yet.   Otherwise, a MariaDBAccount CR is
+	// created here with a generated username as well as a secret with
+	// generated password.   The MariaDBAccount is created without being
+	// yet associated with any MariaDBDatabase.
+	_, _, err = mariadbv1.EnsureMariaDBAccount(
+		ctx, helper, instance.Spec.DatabaseAccount,
```

Comment:
iiuc if the DB password changes there will be a new `DatabaseAccount` with a new name. Is there some situation where the current DatabaseAccount changes and the service operator has to watch this named resource similar to https://github.com/openstack-k8s-operators/keystone-operator/blob/main/controllers/keystoneapi_controller.go#L248-L258 ?

Reply:
the MariaDBAccount object itself, as well as its associated Secret is never meant to change (i.e. the secret is ideally immutable), the mariadbaccount_controller does not respond to changes in the MariaDBAccount/secret itself.  there's only CREATE and DROP for the creation and dropping of the CR.